test/metabase/api/public_test.clj

```diff
@@ -1157,6 +1157,25 @@
               (is (str/starts-with? (:body throttled-response) "Too many attempts!"))
               (is (contains? (:headers throttled-response) "Retry-After")))))))))
 
+(deftest execute-public-dashcard-custom-action-test
+  (mt/with-temp-copy-of-db
+    (perms/revoke-data-perms! (perms-group/all-users) (mt/db))
+    (actions.test-util/with-actions-test-data-and-actions-enabled
+      (mt/with-temporary-setting-values [enable-public-sharing true]
+        (with-temp-public-dashboard [dash {:parameters []}]
+          (actions.test-util/with-action [{:keys [action-id]} {}]
+            (mt/with-temp* [Card [{card-id :id} {:dataset true}]
+                            ModelAction [_ {:slug "custom" :card_id card-id :action_id action-id}]
+                            DashboardCard [{dashcard-id :id} {:dashboard_id (:id dash)
+                                                              :card_id card-id}]]
+              (is (partial= {:rows-affected 1}
```

Comment:
The return value of custom actions is subtly different from that of implicit actions.